syncano/models/fields.py

```diff
@@ -504,11 +505,15 @@ def validate(self, value, model_instance):
         super(ModelField, self).validate(value, model_instance)
 
         if not isinstance(value, (self.rel, dict)):
-            raise self.ValidationError('Value needs to be a {0} instance.'.format(self.rel.__name__))
+            if not self.is_data_object_mixin:
+                raise self.ValidationError('Value needs to be a {0} instance.'.format(self.rel.__name__))
 
         if self.required and isinstance(value, self.rel):
             value.validate()
 
+        if self.is_data_object_mixin and hasattr(value, 'validate'):
```

Comment:
Consider combining it with upper part to always call validate once
`if (self.required and isinstance(value, self.rel)) or (self.is_data_object_mixin and hasattr(value, 'validate')):`